src/serializer/factorify.js

```diff
@@ -92,19 +92,44 @@ function getObjectKeys(obj: BabelNodeObjectExpression): string | false {
 export function factorifyObjects(body: Array<BabelNodeStatement>, factoryNameGenerator: NameGenerator) {
   let signatures = Object.create(null);
 
+  let initializerAstNodeName = "init";
```

Comment:
Can you add comments that illustrate the patterns that you are processing here?
I need some higher-level argument to convince myself that the code transformation is correct.